v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/failureinjectiontesting/SpannerToSourceDbFTBase.java

```diff
@@ -0,0 +1,180 @@
+  }
+
+  public SubscriptionName createPubsubResources(
+      String identifierSuffix,
+      PubsubResourceManager pubsubResourceManager,
+      String gcsPrefix,
+      GcsResourceManager gcsResourceManager) {
+    String topicNameSuffix = "rr-ft" + identifierSuffix;
+    String subscriptionNameSuffix = "rr-ft-sub" + identifierSuffix;
+    TopicName topic = pubsubResourceManager.createTopic(topicNameSuffix);
+    SubscriptionName subscription =
+        pubsubResourceManager.createSubscription(topic, subscriptionNameSuffix);
+    String prefix = gcsPrefix;
+    if (prefix.startsWith("/")) {
+      prefix = prefix.substring(1);
+    }
+    prefix += "/retry/";
+    gcsResourceManager.createNotification(topic.toString(), prefix);
+    return subscription;
+  }
+
+  public PipelineLauncher.LaunchInfo launchRRDataflowJob(
+      String jobName,
+      SpannerResourceManager spannerResourceManager,
+      GcsResourceManager gcsResourceManager,
+      SpannerResourceManager spannerMetadataResourceManager,
+      PubsubResourceManager pubsubResourceManager,
+      String sourceType)
+      throws IOException {
+
+    // create subscription
+    SubscriptionName rrSubscriptionName =
+        createPubsubResources(
+            getClass().getSimpleName(),
+            pubsubResourceManager,
+            getGcsPath("dlq", gcsResourceManager).replace("gs://" + artifactBucketName, ""),
+            gcsResourceManager);
+
+    // Launch Dataflow template
+    FlexTemplateDataflowJobResourceManager flexTemplateDataflowJobResourceManager =
+        FlexTemplateDataflowJobResourceManager.builder(jobName)
+            .withTemplateName("Spanner_to_SourceDb")
+            .withTemplateModulePath("v2/spanner-to-sourcedb")
+            .addParameter("sessionFilePath", getGcsPath("input/session.json", gcsResourceManager))
+            .addParameter("instanceId", spannerResourceManager.getInstanceId())
+            .addParameter("databaseId", spannerResourceManager.getDatabaseId())
+            .addParameter("spannerProjectId", PROJECT)
+            .addParameter("metadataDatabase", spannerMetadataResourceManager.getDatabaseId())
+            .addParameter("metadataInstance", spannerMetadataResourceManager.getInstanceId())
+            .addParameter(
+                "sourceShardsFilePath", getGcsPath("input/shard.json", gcsResourceManager))
+            .addParameter("changeStreamName", "allstream")
+            .addParameter("dlqGcsPubSubSubscription", rrSubscriptionName.toString())
+            .addParameter("deadLetterQueueDirectory", getGcsPath("dlq", gcsResourceManager))
+            .addParameter("maxShardConnections", "5")
+            .addParameter("maxNumWorkers", "1")
+            .addParameter("numWorkers", "1")
+            .addParameter("sourceType", sourceType)
+            .addEnvironmentVariable(
+                "additionalExperiments", Collections.singletonList("use_runner_v2"))
```

Comment:
nit: I think we don't need to explicitly set this anymore with the latest beam version

Reply:
Will remove it in the next PR.